src/internal_events/unix.rs

```diff
@@ -20,7 +20,7 @@ impl InternalEvent for UnixSocketConnectionEstablished<'_> {
 
 #[derive(Debug)]
 pub struct UnixSocketConnectionFailure<'a> {
-    pub error: std::io::Error,
+    pub error: tokio::io::Error,
```

Comment:
It looks like this can be generalized around error type too.

Reply:
Like `UnixSocketError`?

Reply:
Yep.

Reply:
I'm not sure that I understand, but isn't specified type better than generalized? `UnixSocketError` was generalized, because type there can be `std::io::Error` or [tokio_util::codec::LinesCodecError](https://docs.rs/tokio-util/0.3.1/tokio_util/codec/enum.LinesCodecError.html).

Reply:
Oh, that's fine then, nvm! I was worried about consistency, but it doesn't matter that much really.